api/v1alpha1/condition_consts.go

```diff
@@ -37,3 +37,23 @@ const (
 	// DeletedReason (Severity=Info) documents a condition not in Status=True because the underlying object was deleted.
 	DeletedReason = "Deleted"
 )
+
+// Conditions related to the VirtualMachineImages
+const(
+	// VirtualMachineImageOsTypeSupportedCondition denotes that the OS type in the virtual machine image object is
+	// supported by VMService. A VirtualMachineImageOsTypeSupportedCondition is marked true if
+	// - If OS Type is of Linux Family
+	// - If OS Type is supported by hosts in the cluster
+	VirtualMachineImageOsTypeSupportedCondition ConditionType = "VirtualMachineImageOsTypeSupported"
+
+	// VirtualMachineImageSupportedCondition denotes image compatibility with VMService. VMService expects
+	// VirtualMachineImage to be prepared by VI admin for VMService consumption or is a photon image.
+	VirtualMachineImagePreparedForVmServiceCondition ConditionType = "VirtualMachineImagePreparedForVmServiceCondition"
+)
+
+// Condition.Reason for Conditions related to VirtualMachineImages
+const (
+	VirtualMachineImageOsTypeSupportedReason = "VirtualMachineImageOsTypeSupported"
```

Comment:
IMHO, we should not have to set `Reason` for a True condition type. 
https://github.com/kubernetes-sigs/cluster-api/blob/master/util/conditions/setter.go#L109


We can rephrase these (maybe 
`VirtualMachineImageOsTypeUnSupportedReason` and `UnsupportedVirtualmachineImage`) such that we only set these when we have to mark the above conditions as False.

Reply:
https://github.com/vmware-tanzu/vm-operator-api/pull/55#discussion_r568063297
I will let you and @bryanv come to a consensus on what convention to follow

Reply:
fixed